Added msg parameter in wait methods

nikulkarni opened the pull request:
Added msg parameter in wait methods. This will resolve https://github.com/paypal/nemo-view/issues/19

grawk commented on test/methods.js:
```diff
@@ -230,7 +230,7 @@ describe('nemo-view @methods@', function () {
       start = Date.now();
     });
-    nemo.view._waitVisible('#foo.bar.brrao', 3000).then(function (find) {
+    nemo.view._waitVisible('#foo.bar.brrao', 3000, 'Element did not load for specified timeout').then(function (find) {
```
Instead of these changes to the tests, can you create a new test (or tests) to specifically test that the custom message is thrown in the error?

nikulkarni replied:
Oh good call. Actually I found a small bug in drivex, that needs to be merged first though - https://github.com/grawk/selenium-drivex/pull/2

grawk commented:
Can you add the new parameters to the documentation in README?

nikulkarni commented:
addressed comments. Fixed formatting in unit tests as well. So all changes can be seen with `w=1` in the PR url to ignore whitespace changes - https://github.com/paypal/nemo-view/pull/28/files?w=1

Also we would need to merge one more [PR in drivex](https://github.com/grawk/selenium-drivex/pull/4) and publish it before we can merge this PR. Unit tests are bound to fail without it

grawk commented on README.md:
```diff
@@ -252,20 +252,24 @@ The following generic methods are added to `nemo.view`
 `@returns {Promise}` resolves to true or rejected
 
-#### _wait(locatorString[, timeout])
+#### _wait(locatorString[, timeout, msg])
```
May seem like a trivial point, but the syntax above suggests that timeout and msg are both required together. Suggest:

`_wait(locatorString[, timeout[, msg]])

nikulkarni replied:
I didn't understand use of [] to be honest. My guess was whatever that is optional goes into []. Now I understood. will fix it now

grawk commented on test/methods.js:
```diff
@@ -2,280 +2,285 @@
 'use strict';
```
I think the main whitespace change here is that we went from 2 space tabs to 4 space tabs. Would you mind changing the file back to 2 space tabs to avoid this discrepancy? I realize now that the module is internally inconsistent. I personally prefer 2 space tabs and use that as my editor settings. There is not a style guide yet (though there should be for issues such as this). But at least in the meantime let's keep individual files consistent.

grawk commented on test/methods.js:
```diff
@@ -2,280 +2,285 @@
+        nemo.driver.get(nemo.data.baseUrl + '/waits');
+        util.waitForJSReady(nemo);
+        nemo.view.simple.waitButton().click();
+        nemo.view._waitVisible('#outy', 6000, 'Element did not load for specified timeout').getTagName().then(function (tn) {
```
Since we aren't testing the custom message here can we remove it?

grawk commented on test/methods.js:
```diff
@@ -2,280 +2,285 @@
-        done();
-      }
+    it('should find an existing element using the @_wait@positive@ method', function (done) {
+        nemo.view._wait('body', 3000, 'Element did not load for specified timeout').getTagName().then(function (tn) {
```
Since we aren't testing the custom message here can we remove it?

grawk commented on README.md:
```diff
@@ -252,20 +252,24 @@ The following generic methods are added to `nemo.view`
 `@returns {Promise}` resolves to true or rejected
 
-#### _wait(locatorString[, timeout])
+#### _wait(locatorString[, timeout, [msg]])
```
sorry to be trivial yet again, but please put the comma inside the inner [ ]. I.e.
`_wait(locatorString[, timeout[, msg]])`

grawk commented on README.md:
```diff
@@ -252,20 +252,24 @@ The following generic methods are added to `nemo.view`
 `@returns {Promise}` resolves to true or rejected
 
-#### _waitVisible(locatorString[, timeout])
+#### _waitVisible(locatorString[, timeout, [msg]])
```
Please change to:
`_waitVisible(locatorString[, timeout[, msg]])`

nikulkarni commented:
All set. Formatting done with Tab size: 2, Indent: 4, Continuation Indent: 4

grawk commented on README.md:
```diff
@@ -252,20 +252,24 @@ The following generic methods are added to `nemo.view`
 `@returns {Promise}` resolves to true or rejected
 
-#### _wait(locatorString[, timeout])
+#### _wait(locatorString[, timeout, [, msg]])
```
Almost. Consider the brackets denote an optional feature. Thus, the comma before msg must be inside of the brackets, not outside of:
`_wait(locatorString[, timeout[, msg]])`

This is because the comma before msg is only necessary when you actually include a msg parameter

nikulkarni replied:
Done :)

grawk commented:
the build is failing at the jshint step. Can you please resolve any jshint errors?

grawk commented:
Really need to publish a style guide for the nemo modules :/

nikulkarni commented:
ah, sorry for the back and forth and thanks for pushing for these changes. I need to spend more time on nodejs. Too much of Java these days :)

grawk commented:
I think you'll need to rebase to the new develop changes based on the previous PR

nikulkarni commented:
Done, hopefully that was final commit :)

grawk commented on README.md:
```diff
@@ -252,20 +252,24 @@ The following generic methods are added to `nemo.view`
 
 `@argument timeout {Integer} (optional, default 5000)` - ms to wait until rejecting
 
+`@argument msg {String} optional. Message to accompany error in failure case`
```
Slight change:
`@argument msg {String} (optional)` - Message to accompany error in failure case

grawk commented on README.md:
```diff
@@ -252,20 +252,24 @@ The following generic methods are added to `nemo.view`
 
 `@argument timeout {Integer} (optional, default 5000)` - ms to wait until rejecting
 
+`@argument msg {String} optional. Message to accompany error in failure case`
```
Slight change:
`@argument msg {String} (optional)` - Message to accompany error in failure case

grawk commented:
Sorry, ticky tacky docs stuff. But I want to make sure we stay careful and consistent.
